Content.Server/Station/Systems/StationSystem.cs

```diff
@@ -245,10 +237,35 @@ public Filter GetInStation(StationDataComponent dataComponent, float range = 32f
             if (!mapIds.Contains(mapId))
                 continue;
 
+            // Check if the player is directly on any station grid
+            var transform = xform;
+            while (transform.ParentUid != EntityUid.Invalid)
+            {
+                var gridUid = transform.GridUid;
+                if (gridUid != null && dataComponent.Grids.Contains(gridUid.Value))
+                {
+                    filter.AddPlayer(session);
+                    break;
+                }
+
+                transform = xformQuery.GetComponent(transform.ParentUid);
+            }
```

Comment:
An entity's parent can never have a different gridid so not sure the reason to be recursively checking the transform hierarchy?

Reply:
Good point